worlds/AutoWorld.py

```diff
@@ -121,6 +121,7 @@ class World(metaclass=AutoWorldRegister):
     A Game should have its own subclass of World in which it defines the required data structures."""
 
     option_definitions: Dict[str, Option[Any]] = {}  # link your Options mapping
+    options: Dict[str, Option[Any]] = {}  # option names to resulting option object
```

Comment:
this is a static member, i.e. it will have each player of the class use the same options. you should assign a blank dict in `__init__`

If you want it to be a typed dict, this will have to create/overwrite the init per world-class in the metaclass magic, but i'd still fix this behaviour so it's easier to test.